client.go

```diff
@@ -276,7 +276,18 @@ func (cli *Client) DownloadFile(hash string, w io.Writer) (int64, error) {
 		return 0, err
 	}
 	defer resp.Body.Close()
-	return io.Copy(w, resp.Body)
+
+	if resp.StatusCode == http.StatusOK {
+		return io.Copy(w, resp.Body)
+	}
+
+	// See if there is an error in the response.
+	if _, err := cli.parseResponse(resp); err != nil {
+		return 0, err
+	}
+
+	// Last resort return a generic error.
+	return 0, fmt.Errorf("vt: unknown error downloading %s", hash)
```

Comment:
Remove the `vt: ` prefix, for consistency with other error messages, also adding the HTTP error code to the response could be helpful.

Reply:
Done!